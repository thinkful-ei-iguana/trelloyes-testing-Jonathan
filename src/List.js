import React from 'react';
import Card from './Card.js'

function List(props){
  const cardElements = props.cards.map(card => {
    return <Card title={card.title} content={card.content} />
  });

  return (
    <section className="List">
  <header className="List-header">
    <h2>{props.header}</h2>
  </header>
  <div className="List-cards">
      {cardElements}
    <button type="button" className="List-add-button">
      + Add Random Card
    </button>
  </div>
</section>
  )
}

export default List;
