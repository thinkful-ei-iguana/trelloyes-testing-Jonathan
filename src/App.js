import React from 'react';
import Card from './Card.js';
import List from './List.js';
import {STORE} from './store.js';


function App(props) {
  const listElements = props.store.lists.map(list => {
    const cards = list.cardIds.map(id => {
      return props.store.allCards[id]
    } )
    return <List header={list.header} cards={cards} />
  });
  return (
    <main className="App">
      <header className="App-header">
        <h1>Trelloyes!</h1>
      </header>
      <div className="App-list">
      {listElements}
      </div>
    </main>
  );
}
export default App;