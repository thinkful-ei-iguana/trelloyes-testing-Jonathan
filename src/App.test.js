import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Card from './Card.js';
import List from './List.js';
import renderer from 'react-test-renderer';

//smoke test for Card
describe(Card, () => {
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');

      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<Card />, div);

      // Clean up
      ReactDOM.unmountComponentAtNode(div);
  });
});

//smoke test for List
describe(List, () => {
  it('renders without crashing', () => {
      // Create a DOM element to render the component into
      const div = document.createElement('div');

      // Render the component
      // If something is wrong it will fail here
      ReactDOM.render(<List cards={[{title: 'this is the title', content: 'this is content'}]} />, div);

      // Clean up
      ReactDOM.unmountComponentAtNode(div);
  });
});

//snapshot test for Card
describe('<Card />', () => {
  it('renders this UI as expected', () => {
      // Render the component, as JSON
      const tree = renderer.create(<Card />).toJSON();
      // Check whether it matches the previous snapshot
      // Stored in __snapshots__/App.test.js.snap
      expect(tree).toMatchSnapshot(); 
  });
});


//snapshot test for List
describe('<List />', () => {
  it('renders this UI as expected', () => {
      // Render the component, as JSON
      const tree = renderer.create(<List cards={[{title: 'this is the title', content: 'this is content'}]} />).toJSON();
      // Check whether it matches the previous snapshot
      // Stored in __snapshots__/App.test.js.snap
      expect(tree).toMatchSnapshot(); 
  });
});

