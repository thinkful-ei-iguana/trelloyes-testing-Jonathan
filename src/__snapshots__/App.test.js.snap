// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders this UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        this is the title
      </h3>
      <p>
        this is content
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
